zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -308,6 +308,48 @@ public synchronized boolean append(TxnHeader hdr, Record txn, TxnDigest digest)
         return true;
     }
 
+    @Override
+    public synchronized boolean append(Request request) throws IOException {
```

Comment:
We're duplicate code here.

Perhaps we can use one `append(Request request)` for all the overloads and changes the other usage (which are all test usage I saw).